### ISSUE TITLE
A generic not being `instanceof Object` is incorrectly reported as type `never`

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [microsoft/TypeScript#13327]
**TypeScript Version:**  2.1.1

**Code**

```typescript
class Test<T> {
    public test(key: T) {
        if (!(key instanceof Object)) {
            key.toString(); // Property 'toString' does not exist on type 'never'
        }
    }
}
```

**Expected behavior:**

It's possible for `key instanceof Object` to be false, such as if `key` is a primitive. There should be no error.

**Actual behavior:**

`key` is thought to be of type `never`. It looks like TypeScript thinks the generic must be a non-primitive Object descendant?

Note that this is unique to generics; using a regular `any` type in an `instanceof` gives no error.

```typescript
class PassingTest {
    public test(key: any): void {
        if (!(key instanceof Object)) {
            key.toString();
        }
    }
}
```

**Comment by RyanCavanaugh** [comment]
This has been fixed
